ansible/playbooks/roles/certbot_auto/tasks/main.yml

```diff
@@ -0,0 +1,41 @@
+---
+- name: gather Cloud/OS specific variables
+  include_vars: '{{ item }}'
+  with_first_found:
+    - '{{ ansible_bios_version }}-{{ ansible_distribution }}.yml'
+    - '{{ cloud_platform }}-{{ ansible_distribution }}.yml'
+
+- name: install certbot-auto
+  become: true
+  copy:
+    src: 'certbot-auto'
+    dest: '/usr/local/bin/'
+    mode: '0755'
+    owner: '{{ USER.user }}'
+    group: '{{ USER.user }}'
+
+- name: create ssl certificate
+  shell: 'certbot-auto certonly --standalone --email dev@harmony.one -d {{ domain }} -q --agree-tos'
```

Comment:
regarding letsencrypt operations, it is highly recommended to --dry-run first and make sure the operation succeeds before actually trying, because the cert generation/renewal can fail due to external factors (like the domain issue, etc.) Also, for cert generation/renewal, port 80 should be running which means you need to have the server running and listening to port 80 for response upon letsencrypt opepration.

Can you implement such letsencrypto operations to be dry-run before the actual operation, and proceed without dry-run only after checking the success of the dry run.

Reply:
yeah. this is the first step for this playbook. and port 80 not used now. i tried this command with --dry-run also actually trying it works.